src/AccountManager.mo

```diff
@@ -331,54 +250,57 @@ module {
       return ?(inc, creditInc);
     };
 
-    // This function is async* but is guaranteed to never throw.
-    func processAllowance(p : Principal, account : ICRC1.Account, amount : Nat) : async* DrawResponse {
-      let res = await* Ledger.draw(p, account, amount);
-      if (R.isOk(res)) {
-        totalConsolidated_ += amount;
-        issue(p, amount);
+    // Proccesses allowance.
+    func processAllowance(p : Principal, account : ICRC1.Account, amount : Nat, expectedFee : ?Nat) : async* DepositFromAllowanceResponse {
+      switch (expectedFee) {
+        case null {};
+        case (?f) if (f != fee(#allowance)) return #err(#BadFee { expected_fee = fee(#allowance) });
       };
-      res;
-    };
 
-    // This function is async* but is guaranteed to never throw.
-    func depositFromAllowanceRecursive(p : Principal, account : ICRC1.Account, amount : Nat, retry : Bool) : async* DepositFromAllowanceResponse {
-      if (amount < minimum(#deposit)) return #err(#TooLowQuantity);
+      if (amount <= fee(#allowance)) return #err(#TooLowQuantity);
 
-      let res = await* processAllowance(p, account, amount);
+      let res = await* Ledger.draw(p, account, amount);
 
-      // catch BadFee
       switch (res) {
+        case (#ok txid) #ok(amount - fee(#allowance), txid);
```

Comment:
This return value `amount - fee(@#allowance)` is useless. The calling code is ignoring it anyway. And if the calling code needed it then it could calculate it itself from the arguments because it is equal to `amount - expectedFee` (if #ok is returned).